specs/form-associated-custom-element.md

```diff
@@ -0,0 +1,114 @@
+# Form Associated Custom Elements
+
+## Overview
+This spec regards a general implementation to facilitate form communication and connection for components that should be form-associated, including checkboxes, radios, text inputs, and other components which store user input and should be captured during form submission.
+
+### Background
+Components that replace or behave like native form elements (input, textarea, select) should generally behave like native like their native counterpart. One key aspect to this is associating the element with the parent form. To do this, we will expose a mechanism to expose form-association to custom elements that can be shared across implementations with maximum re-use.
+
+### Use Cases
+Any custom element that should associate a value to a form.
+
+## Design
+The implementation will be an *abstract class* that will extend `FastElement`. The class will expose implementations for all of the common form element capabilities. It will also expose and manage an implementation for when the form-associated custom element APIs are not available.
+
+Another option I looked at is using a decorator, but there are a few challenges with that approach:
+  1. decorators cannot augment the class type directly, so we would need to use a pattern similar to https://www.typescriptlang.org/docs/handbook/mixins.html to gain accurate type definitions
+  2. Decorators cannot (easily?) provide a *default* implementation that can be overridden by the extending class. Using an abstract class allows straight overrides of implementation or merging of implementation with super
+
+The implementation will standardize an interface between two common cases: browsers with form-associated-custom-element support and browsers without.
+
+### Browsers with FACE support
+For browsers *with* support, methods and properties will generally proxy to the `ElementInternals` implementation.
+
+### Browsers with FACE support
```

Comment:
This title is duplicated, also, can we get a link to what is FACE support?

Reply:
I can define - "form associated custom elements"